### ISSUE TITLE
Is it possible to write text into a specific field?

### ISSUE THREAD

**Issue by LaithAzer** [gettalong/hexapdf#22]
Hi,

I was hoping you can help me with my issue. I tried going through the documentation and API, and I think I just don't understand the PDF specs well enough, but is it possible to use this gem to write into text fields like the one in this pdf? http://foersom.com/net/HowTo/data/OoPdfFormExample.pdf

I want to basically load a pdf, find all the fields and be able to write certain text into each of them.

Thanks,
Laith

**Comment by gettalong** [comment]
**Tl;dr**: Not currently possible but on TODO list.

**Slightly longer version:**

You *could* already do this if you dove into the PDF spec section 12.7 "Interactive Forms" and manipulated the applicable objects of the PDF file. However, there is currently no built-in support for doing this, i.e. there are currently no convenience methods like e.g. with images (`HexaPDF::Document#images`) that allow processing forms in an easy and straightforward manner.

There are also *two* ways to store forms in a PDF file since version 1.5, AcroForms and XFA forms. Since XFA forms are deprecated since version 2.0 (which came out earlier this year) I will probably only focus on AcroForms and I guess most PDF forms use AcroForms and not XFA forms (the latter are mostly used for very complex PDF forms, like those the EU uses for project applications).

I have asked a while ago on which features I should work next (see https://www.patreon.com/posts/next-features-15075087; alas there wasn't much feedback) and this was on my TODO list. There are, however, some other things I will implement first so this might take a while until it lands in HexaPDF.
